Sources/SwiftDocC/Utility/SemanticVersion+Comparable.swift

```diff
@@ -11,7 +11,13 @@
 import Foundation
 import SymbolKit
 
-extension SymbolGraph.SemanticVersion: Comparable {
+// If SymbolKit adds Comparable conformance it's reasonable to expect that its behavior would be compatible.
+//
+// If a future SymbolKit implementation considers the "prerelease" and "buildMetadata" components _after_ the
+// "major", "minor", and "patch" components, that remains compatible with the behavior that SwiftDocC expects.
+//
+// Since `@retroactive` is new in the Swift 6 compiler, use the backwards compatible syntax (fully-qualified types) to declare retroactive conformance.
+extension SymbolKit.SymbolGraph.SemanticVersion: Swift.Comparable {
```

Comment:
It would be a bit clearer to start this comment with an explanation of what the comment is addressing, otherwise out of context I think it'll be hard to figure out that this relates to conformances of external types. 
```suggestion
// Use fully-qualified names to silence a warning about retroactive conformance.
//
// Since `@retroactive` is new in the Swift 6 compiler, use the backwards compatible syntax (fully-qualified types) to declare retroactive conformance.
//
// If SymbolKit adds Comparable conformance it's reasonable to expect that its behavior would be compatible.
//
// If a future SymbolKit implementation considers the "prerelease" and "buildMetadata" components _after_ the
// "major", "minor", and "patch" components, that remains compatible with the behavior that SwiftDocC expects.
extension SymbolKit.SymbolGraph.SemanticVersion: Swift.Comparable {
```

This also applies to some other comments in this PR which follow the same format.

Reply:
Updated in https://github.com/apple/swift-docc/pull/957/commits/eae4c828b2ccf3218320a8a05c6f572d2dab86d1